pykafka/balancedconsumer.py

```diff
@@ -259,6 +268,11 @@ def stop(self):
             # facilities for that in ChildrenWatch - as a workaround we check
             # self._running in the watcher callbacks (see further down)
 
+        # Unregister the cleanup handler
+        unregister_cleanup_func(self._cleanup_func, self)
```

Comment:
One way would be to have `register_cleanup_func` wrap `self` in a `weakref.proxy` and wrap `self._cleanup_func` in a thin wrapper that just catches and ignores `ReferenceError`. That makes it all awfully ugly to read and understand, but I don't know a better option for this.